### PR TITLE
fixed item loading issue in case the item factory gets added late

### PR DESCRIPTION
**Pull request by sjka**
Signed-off-by: Simon Kaufmann <simon.kfm@googlemail.com>

### CODE REVIEW

File: bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/items/ManagedItemProvider.java [hunk 1]
```diff
@@ -123,6 +176,26 @@ private String toItemFactoryName(Item item) {
...
+            // retry failed creation attempts
+            Iterator<ItemInfo> iterator = failedToCreate.iterator();
+            while (iterator.hasNext()) {
+                ItemInfo itemInfo = iterator.next();
+                Item item = itemFactory.createItem(itemInfo.itemType, itemInfo.itemName);
+                if (item != null) {
+                    iterator.remove();
+                    notifyListenersAboutAddedElement(item);
+                } else {
+                    logger.debug("The added item factotry '{}' still could not instantiate item '{}'.", itemFactory,
```
Comment (maggu2810) [inline bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/items/ManagedItemProvider.java]: typo: factotry

File: bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/items/ManagedItemProvider.java [hunk 2]
```diff
@@ -56,6 +57,58 @@
...
     private Collection<ItemFactory> itemFactories = new CopyOnWriteArrayList<ItemFactory>();
 
+    private final Set<ItemInfo> failedToCreate = new HashSet<>();
+
+    private static class ItemInfo {
+        private final String itemType;
+        private final String itemName;
+
+        public ItemInfo(String itemType, String itemName) {
+            super();
```
Comment (maggu2810) [inline bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/items/ManagedItemProvider.java]: It is perhaps just my personal meaning, but we could drop the call to the default constructor of the parent.

File: bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/items/ManagedItemProvider.java [hunk 3]
```diff
@@ -123,6 +176,26 @@ private String toItemFactoryName(Item item) {
...
+                    iterator.remove();
+                    notifyListenersAboutAddedElement(item);
+                } else {
+                    logger.debug("The added item factotry '{}' still could not instantiate item '{}'.", itemFactory,
+                            itemInfo.itemName);
+                }
+            }
+
+            if (failedToCreate.isEmpty()) {
+                logger.info("Finished loading all items that could not have been created before.");
```
Comment (maggu2810) [inline bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/items/ManagedItemProvider.java]: loading _of_ all (?)

**Comment by maggu2810** [comment]
LGTM, some minor remarks.

**Comment by sjka** [comment]
@maggu2810 I have incorporated your comments, thanks for the feedback!

**Pull request merged**